pkgs/top-level/all-packages.nix

```
@@ -16367,7 +16367,7 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) CoreServices;
   };
 
-  buildBazelPackage = callPackage ../build-support/build-bazel-package { };
+  buildBazelPackage = (if stdenv.isDarwin then darwin.apple_sdk_11_0.callPackage else callPackage) ../build-support/build-bazel-package { };
```

borrowed this fix from https://github.com/NixOS/nixpkgs/pull/198701#discussion_r1009044255
It fixes `bazel-watcher` on x86_64-darwin, but only if placed here, changing `callPackage` on `bazel-watcher` itself doesn't help

---

This is wrong in the PR and shouldn't be used like this.

```suggestion
  buildBazelPackage = darwin.apple_sdk_11_0.callPackage ../build-support/build-bazel-package { };
```

---

Looked more logical that for -linux it shouldn't be using darwin.*.callPackage, but will apply :+1:

---

updated and squashed commits together

---

The callPackage is a wrapper that for linux is a noop.

---

I don't doubt that it currently works either way. But I can imagine there being a darwin.apple_sdk_11_0 wrapper, and say openbsd.something-something or whatever that own act only on their specific systems and usecases. And then which should be used if package needs both tweaks, not sure how to compose them. And not sure how to improve readability for default case, we aren't going to use apple_sdk on linux, but just looking at `darwin.apple_sdk_11_0.callPackage` there could be impression that we would / that it is a darwin-only package.

But anyway, I applied the suggestion and this general improvements stuff goes a little beyond the scope of bazel_5 version bump :slightly_smiling_face: